src/hooks/useLatest.ts

```diff
@@ -0,0 +1,11 @@
+import { useInsertionEffect, useRef } from 'react';
+
+export function useLatest<T>(value: T) {
+    const ref = useRef(value);
+
+    useInsertionEffect(() => {
```

Comment:
Why do you use useInsertionEffect instead of useEffect/useLayoutEffect?

Reply:
Callback refs are called before useLayoutEffect, which can lead to an irrelevant values in the callback refs.

useInsertionEffect is called before the callback refs, which guarantees the relevance of the values.